File: beakjoon-c,cpp/현수막.cpp
#include<iostream>
#include<array>
#include<queue>

using namespace std;
typedef pair<int,int> point;
array<array<int,251>,251> check;
queue<point> q;
int dx[8] = {-1,1,0,0,1,-1,1,-1},dy[8] = {0,0,1,-1,-1,1,1,-1};
int m,n;
int BFS(int x,int y)
{
	check[x][y] = 0;
	q.push({x,y});
	while(!q.empty())
	{
		x = q.front().first;
		y = q.front().second;
		q.pop();
		for(int i = 0; i < 8; i++)
		{
			int px = x + dx[i];
			int py = y + dy[i];
			if(0 > px || px >= m || 0 > py || py >= n)
				continue;
			
			if(!check[px][py])
				continue;
			else
			{
				check[px][py] = 0;
				q.push({px,py});
			}
		}
	}
	return 1;
} 
int main()
{
	scanf("%d %d",&m,&n);	
	for(int i = 0; i < m; i++){
		for(int j = 0; j < n; j++){
			scanf("%d",&check[i][j]);
		}
	}
	
	int count = 0;
	for(int i = 0; i < m; i++){
		for(int j = 0; j < n; j++){
			if(check[i][j])
				count += BFS(i,j);
		}
	}
	printf("%d",count);
	return 0;
}

File: beckjoon-pyhton/현수막.py
def BFS(check,x,y,m,n):
    check[x][y] = 0
    q = [[x,y]]
    dx = [1,-1,0,0,1,-1,1,-1]
    dy = [0,0,1,-1,1,-1,-1,1]
    while(len(q)):
        x = q[0][0]
        y = q[0][1]
        q.pop(0)
        for i in range(8):
            px = x + dx[i]
            py = y + dy[i]
            if 0 > px or px >= m or 0 > py or py >= n:
                continue
            if not check[px][py]:
                continue
            else:
                check[px][py] = 0
                q.append([px,py])
    return 1

def main():
    m,n = map(int,input().split())
    check = [[0]*n for _ in range(m)]
    for i in range(m):
        x = list(map(int,input().split()))
        check[i] = x
    
    count = 0
    for i in range(m):
        for j in range(n):
            if check[i][j]:
                count += BFS(check,i,j,m,n)
    print(count)
    return
main()